nobl9/resource_agent.go

```diff
@@ -3,254 +3,159 @@ package nobl9
 import (
 	"context"
 	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
 	n9api "github.com/nobl9/nobl9-go"
 )
 
-//nolint:lll
+const agentTypeKey = "agent_type"
+
 func resourceAgent() *schema.Resource {
 	return &schema.Resource{
-		Schema: map[string]*schema.Schema{
-			"name":         schemaName(),
-			"display_name": schemaDisplayName(),
-			"project":      schemaProject(),
-			"description":  schemaDescription(),
-
-			"source_of": {
-				Type:        schema.TypeList,
-				Required:    true,
-				MinItems:    1,
-				MaxItems:    2,
-				Description: "Source of Metrics and/or Services",
-				Elem: &schema.Schema{
-					Type:        schema.TypeString,
-					Description: "Source of Metrics or Services",
-				},
-			},
+		Schema:        agentSchema(),
+		CreateContext: resourceAgentApply,
+		UpdateContext: resourceAgentApply,
+		DeleteContext: resourceAgentDelete,
+		ReadContext:   resourceAgentRead,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Description: "[Agent configuration documentation](https://docs.nobl9.com/the-nobl9-agent)",
```

Comment:
It's actually a funny thing, because this docs URL points to some weird cached version of docs, that doesn't include GCM and Influx (or dark mode!). We should update it and re-generate docs.

```suggestion
		Description: "[Agent configuration documentation](https://docs.nobl9.com/nobl9_agent)",
```